game/src/sandbox/dashboards/commuter.rs

```diff
@@ -320,15 +355,66 @@ impl State for CommuterPatterns {
                 )
                 .named("scale");
                 self.composite.replace(ctx, "scale", new_scale);
-            } else {
-                self.composite.replace(
-                    ctx,
-                    "current",
-                    "None selected".draw_text(ctx).named("current"),
-                );
             }
+        } else {
+            self.composite.replace(
+                ctx,
+                "current",
+                "None selected".draw_text(ctx).named("current"),
+            );
         }
 
+        // Draw outline for Locked Selection
+        match self.selection {
+            BlockSelection::Locked(clicked, _) => {
+                let key = (clicked, BlockDrawableType::BaseLocked);
+                if !self.block_drawables.contains_key(&key) {
+                    let d = ctx.upload(GeomBatch::from(vec![(
+                        Color::BLACK,
+                        self.blocks[clicked]
+                            .shape
+                            .to_outline(Distance::meters(10.0))
+                            .unwrap(),
+                    )]));
+                    self.block_drawables.insert(key, d);
+                }
+            }
+            _ => {}
+        };
+
+        // While selection is locked, draw an overlay with compare_to information for the hovered
+        // block
+        match self.selection {
+            BlockSelection::Locked(clicked, Some(hovered)) => {
+                let key = (hovered, BlockDrawableType::CompareTo);
+                if !self.block_drawables.contains_key(&key) {
+                    let block = &self.blocks[hovered];
+
+                    let border = block.shape.to_outline(Distance::meters(10.0)).unwrap();
+
+                    let mut batch = GeomBatch::from(vec![(Color::WHITE.alpha(0.8), border)]);
+
+                    let base_block = &self.blocks[clicked];
+                    let others = self.count_per_block(&base_block);
+                    let count = others
+                        .into_iter()
+                        .find(|(block, _)| block.id == hovered)
+                        .map(|(_, count)| count)
+                        .unwrap_or(0);
+                    let label_text = format!("{}", count);
+                    batch.append(
+                        Text::from(Line(label_text).fg(Color::WHITE))
```

Comment:
I'm having trouble reading this even on the pink blocks. How about `Color::BLACK` and before the `centered_on`, do `.scale(2.0)` or so? That looks a little easier to read to me, personally.

Reply:
Done!

Here's what that looks like:

<img width="1792" alt="Screen Shot 2020-07-24 at 5 37 41 PM" src="https://user-images.githubusercontent.com/217057/88443152-6df8bc80-cdd4-11ea-8a5d-f8f5844126ef.png">

<img width="1792" alt="Screen Shot 2020-07-24 at 5 37 48 PM" src="https://user-images.githubusercontent.com/217057/88443148-69cc9f00-cdd4-11ea-981f-159b77cdbee2.png">

I think it's better!